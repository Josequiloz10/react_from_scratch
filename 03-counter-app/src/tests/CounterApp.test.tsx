import {describe, expect, test} from "vitest"
import {shallow} from 'enzyme'
import CounterApp from "../CounterApp";

describe('testing CounterApp', () => {
    let wrapper = shallow(<CounterApp />);

    //@ts-ignore
    beforeEach(() => {
        wrapper = shallow(<CounterApp />)
    })

    test('it should show CounterApp with default values', () => {
        expect(wrapper).toMatchSnapshot();
    })

test('it should show CounterApp with default 100 values', () => {

    const wrapper = shallow(
        <CounterApp
            value={100}
        />)

    const valueH2 = wrapper.find('h2').text().trim();
    expect(valueH2).toBe('100');
})

test('it should increse with +1 button', () => {

    wrapper.find('button').at(0).simulate('click');
    const valueH2 = wrapper.find('h2').text().trim();
    expect(valueH2).toBe('11');
})

test('it should decrease with -1 button', () => {

    wrapper.find('button').at(2).simulate('click');
    const valueH2 = wrapper.find('h2').text().trim();
    expect(valueH2).toBe('9');
})

test('it should restart the value', () => {
    const wrapper = shallow(
        <CounterApp
            value={105}
        />)
    wrapper.find('button').at(0).simulate('click');
    wrapper.find('button').at(0).simulate('click');
    wrapper.find('button').at(1).simulate('click');
    const valueH2 = wrapper.find('h2').text().trim();
    expect(valueH2).toBe('105');
})

})
