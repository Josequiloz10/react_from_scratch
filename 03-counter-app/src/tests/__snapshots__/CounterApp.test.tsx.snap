// Vitest Snapshot v1

exports[`testing CounterApp > it should show CounterApp with default values 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
